machine/views/machine_instance_views.xml

```diff
@@ -6,11 +6,14 @@
         <field name="model">machine.instance</field>
         <field name="mode">primary</field>
         <field name="arch" type="xml">
-            <!-- TODO: might be better to have archive button on both forms.
-            Though when archiving template, it should archive its instances
-            too. -->
             <form>
                 <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <button name="toggle_active"
+                            type="object" class="oe_stat_button" icon="fa-archive">
+                            <field name="active" widget="boolean_button" options="{&quot;terminology&quot;: &quot;archive&quot;}"/>
```

Comment:
I suggest to add a confirmation message something like in projects (e.g. confirm="(Un)archiving a project automatically (un)archives its tasks. Do you want to proceed?").

Reply:
Maybe, but its good enough. Its easy to toggle back. Not that it will make a change with no return:)